src/main/scala/com/gu/fastly/Lambda.scala

```diff
@@ -1,45 +1,55 @@
 package com.gu.fastly
 
-import org.apache.commons.codec.digest.DigestUtils
-import com.amazonaws.services.kinesis.clientlibrary.types.UserRecord
+import java.io.IOException
+
 import com.amazonaws.services.kinesis.model.Record
+import com.amazonaws.services.lambda.runtime.{Context, RequestHandler}
 import com.amazonaws.services.lambda.runtime.events.KinesisEvent
+import com.gu.crier.model.event.v1.{Event, EventPayload, EventType}
+import com.gu.fastly.CrierEventProcessor.{decodeRecord, successfulEvents}
 import okhttp3._
-import com.gu.crier.model.event.v1._
+import org.apache.commons.codec.digest.DigestUtils
+
 import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{ Future, Promise, Await, duration }
 
-class Lambda {
+class Lambda extends RequestHandler[KinesisEvent, Unit] {
 
   private val config = Config.load()
   private val httpClient = new OkHttpClient()
 
-  def handle(event: KinesisEvent) {
+  def handleRequest(event: KinesisEvent, context: Context): Unit = {
+    Await.result(processEvent(event), duration.Duration(context.getRemainingTimeInMillis, duration.MILLISECONDS))
```

Comment:
This is where we await the result of the future for the time available to the Lambda

Reply:
maybe you want to finish a bit before the lambda shuts down so you can log something?

Reply:
Hmm yes, or I could move the logging to the end of the processEvent function